ygot/render.go

```diff
@@ -1025,6 +1042,28 @@ func Marshal7951(d interface{}, args ...Marshal7951Arg) ([]byte, error) {
 	return js, nil
 }
 
+func hasBelongingModule(s GoStruct) bool {
+	return reflect.ValueOf(s).MethodByName("ΛBelongingModule").IsValid()
+}
+
+// belongingModuleName returns the name of the belonging module of the input GoStruct.
```

Comment:
If we added this to the `ValidatedGoStruct` interface, I think that you could then just call this directly. I guess the downside of doing this is that users need to regenerate their bindings code at the same time as upgrading, but it seems cleaner in general.  (The argument to say that bindings should be regenerated is that this defect is not resolved unless they both regenerate the code and upgrade ygot).

Reply:
I'm thinking that `ΛBelongingModule` is not required by validation. If so then should it still be part of the interface definition?

Reply:
Good question -- `ValidatedGoStruct` was created because at some point we had users that didn't care about validation, and `GoStruct` needed to coexist with it. At this point, I'm not sure that logic is correct -- essentially someone working with a `ValidatedGoStruct` really probably cares whether they are working with a type generated by ygot or not. If this is the intention of the interface (rather than it being intended to show that this is a validatable struct) then I'd suggest extending it for reasons that don't immediately link to validation seems fine.

That being said, I'm not sure `ValidatedGoStruct` today covers all the helper methods that we would generate by default.

WDYT here?

Reply:
I've always been confused by this interface so thanks for clarifying. I agree to add to the interface, it looks like the purpose of this interface is to expose more methods to the user rather than for validation only.

It makes sense that the user should always be using the same version of ygot to generate and operate on the GoStructs, so I don't see it as a problem that we require users to update both at the same time. The two are coupled and make assumptions about each other and this mutual dependency is hard to break and not necessary to break.

Reply:
SGTM :-)

If we're agreeing with this - we should perhaps add any method that we /always/ generate to this interface, and perhaps update its description to just say that it is implemented by any ygot-generated struct.

We could probably say that `GoStruct` is deprecated at this point.

Reply:
Done.